meeg_preprocessing/utils.py

```diff
@@ -151,15 +153,28 @@ def setup_provenance(script, results_dir, config=None, use_agg=True):
     - sets log file for sterr output
     - writes log file with runtime information
     """
+    import inspect
+    if script is None:
+        script = inspect.stack()[1][1]
+        if script[0] == '<':
+            script = '__main__'
```

Comment:
Because Matlab is designed inefficently ;) This is why you have explicit name spaces in Python. You can selectively import isolated code from files. As a consequence of Python's optimization that help achieving this efficiently you loose changes during run time. This is btw. a very good feature. Imaging the code changes while you're running an anlyis, for example because your cat is editing it.